Tutorials/a1_ros_setup.md

```diff
@@ -0,0 +1,102 @@
+
+While Parallels does not share the details of how `Shared` networking operates, I have successfully configured it so that ROS installed in the VM can talk to TX2 directly. Here are the steps I followed:
+1. Manually set Mac-Mini's ip to be within the same subnet of the TX2 as `192.168.123.xx`, so that you can ping/ssh into TX2 from your Mac-Mini. 
+2. Log in to VM, and check that you can ping TX2's IP (`192.168.123.12`) and ssh into it (`ssh unitree@192.168.123.12`) directly from **inside** the VM. This should work automatically as long as your previous step is configured correctly. Also, your `ssh` activity to TX2 should be recorded as coming from the Mac-Mini's ip.
+3. Add the following environment variables to `~/.bashrc` **inside** the VM:
+```
+export ROS_MASTER_URI=http://192.168.123.12:11311
+```
+and check that running `rostopic list` can identify TX2's ROS topics.
+4. Publish some messages on TX2, and check whether you can receive them on M1 by running `rostopic echo`. If you can see the topic name, but cannot see the actual messages being published, try adding TX2's hostname to the VM's `/etc/hosts` file, or see [this post](https://answers.ros.org/question/90536/ros-remote-master-can-see-topics-but-no-data/) for more helps in trouble-shooting.
+
+## Make ROS processes run at boot time
+Since the robot can suffer from unexpected conditions and stop/reboot randomly esp. during outdoor operation, it can be painful to re-connect to the robot every time and manually start the ROS processes. There are several ways to make scripts run at start-up in Ubuntu, including [`init.d`](https://en.wikipedia.org/wiki/Init), [`systemd`](https://en.wikipedia.org/wiki/Systemd), [`cron` @reboot](https://phoenixnap.com/kb/crontab-reboot) and [`autostart`](https://arcolinux.com/how-to-autostart-any-application-on-any-linux-desktop/). They operate at different levels. At the abstract level, `init.d` runs at OS bootup. `systemd` runs as a service when OS is running. `cron` is a `systemd` service that allows more flexible scheduling. `autostart` is Ubuntu-specific that's tied to Gnome.
+
+Since ROS requires networking, putting it in `init.d` requires careful ordering and can be tricky. `autostart` and `cron` don't have good support on defining custom environment variables or sourcing other bash scripts, which is required for ROS. Therefore, I have found `systemd` to be the easiest way to configure this.
```

Comment:
This is very helpful info... I've always run into arbitrary logic for which to choose.